sympy/physics/continuum_mechanics/beam.py

```diff
@@ -0,0 +1,437 @@
+            self._boundary_conditions['slope'].extend([s_bcs for s_bcs in slope])
+        if deflection:
+            self._boundary_conditions['deflection'].extend([d_bcs for d_bcs in deflection])
+
+    def apply_moment_boundary_conditions(self, *m_bcs):
+        """
+        Takes only the moment boundary conditions as input.
+
+        Examples
+        ========
+        >>> from sympy.physics.continuum_mechanics.beam import Beam
+        >>> from sympy import Symbol
+        >>> E = Symbol('E')
+        >>> I = Symbol('I')
+        >>> b = Beam(4, E, I)
+        >>> b.apply_boundary_conditions(moment=[(0, 4), (4, 0)], deflection=[(0, 2)], slope=[(0, 1)])
+        >>> b.boundary_conditions
+        {'deflection': [(0, 2)], 'moment': [(0, 4), (4, 0)], 'slope': [(0, 1)]}
+        >>> b.apply_moment_boundary_conditions((4, 3), (5, 0))
+        >>> b.boundary_conditions['moment']
+        [(0, 4), (4, 0), (4, 3), (5, 0)]
+
+        """
+        for bcs in m_bcs:
+            self._boundary_conditions['moment'].append(bcs)
+
+    def apply_slope_boundary_conditions(self, *s_bcs):
+        """
+        Takes only the slope boundary conditions as input.
+
+        Examples
+        ========
+        >>> from sympy.physics.continuum_mechanics.beam import Beam
+        >>> from sympy import Symbol
+        >>> E = Symbol('E')
+        >>> I = Symbol('I')
+        >>> b = Beam(4, E, I)
+        >>> b.apply_boundary_conditions(moment=[(0, 4), (4, 0)], deflection=[(0, 2)], slope=[(0, 1)])
+        >>> b.boundary_conditions
+        {'deflection': [(0, 2)], 'moment': [(0, 4), (4, 0)], 'slope': [(0, 1)]}
+        >>> b.apply_slope_boundary_conditions((4, 3), (5, 0))
+        >>> b.boundary_conditions['slope']
+        [(0, 1), (4, 3), (5, 0)]
+
+        """
+        for bcs in s_bcs:
+            self._boundary_conditions['slope'].append(bcs)
+
+    def apply_deflection_boundary_conditions(self, *d_bcs):
+        """
+        Takes only the slope boundary conditions as input.
+
+        Examples
+        ========
+        >>> from sympy.physics.continuum_mechanics.beam import Beam
+        >>> from sympy import Symbol
+        >>> E = Symbol('E')
+        >>> I = Symbol('I')
+        >>> b = Beam(4, E, I)
+        >>> b.apply_boundary_conditions(moment=[(0, 4), (4, 0)], deflection=[(0, 2)], slope=[(0, 1)])
+        >>> b.boundary_conditions
+        {'deflection': [(0, 2)], 'moment': [(0, 4), (4, 0)], 'slope': [(0, 1)]}
+        >>> b.apply_deflection_boundary_conditions((4, 3), (5, 0))
+        >>> b.boundary_conditions['deflection']
+        [(0, 2), (4, 3), (5, 0)]
+
+        """
+        for bcs in d_bcs:
+            self._boundary_conditions['deflection'].append(bcs)
+
+    def apply_load(self, value, start, order):
+        x = self.variable
+        value = sympify(value)
+        start = sympify(start)
+        order = sympify(order)
+
+        self._load += value*SingularityFunction(x, start, order)
+
+    @property
+    def load(self):
```

Comment:
I think the idea of doing the following will not be that good.
```py
b.load += -3*SingularityFunction(x, 0, -1)
```
Because
- We can't test whether the input is a SingularityFunction object. The one which is passed here is Mul object.
- We have one setter function know as apply_load.

I think making load as a property would be the only good idea. Let's leave the idea of making it as setter too.